http/dotnet/httpclient/src/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -24,9 +24,9 @@ public static class HttpRequestMessageExtensions
         /// <returns>A request option</returns>
         public static T GetRequestOption<T>(this HttpRequestMessage httpRequestMessage) where T : IRequestOption
         {
-            if(httpRequestMessage.Options.TryGetValue(
-                new HttpRequestOptionsKey<IRequestOption>(typeof(T).FullName),
-                out IRequestOption requestOption))
+            if(httpRequestMessage.Properties.TryGetValue(
```

Comment:
the fact we have to revert to an [obsolete property](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.properties?view=net-5.0#System_Net_Http_HttpRequestMessage_Properties) bothers me a little.
Could we imagine multi-targeting net5.0 and use conditional build here? how much more work would that be for us?

Reply:
Right now, I can only see a few mentions of the property and we could probably do it. It however gets tricky going forward as one would need to keep remembering to add the build flags whenever using the property and this might be a maintenance pain.
I also looked at the source and they both point to the same collection (see [here](https://github.com/dotnet/runtime/blob/e64cce6fbbde72d1ea61010bd08f11ba7b51afc9/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs#L116)) so that might cause problems for ourselves unnecessarily.

Reply:
ahh, good thinking to check the implementation! yeah probably more trouble than it's worth then. Thanks!